src/Validation/Validation.php

```diff
@@ -839,7 +827,7 @@ public static function equalTo($check, $comparedTo): bool
      * @param array<string> $extensions file extensions to allow. By default extensions are 'gif', 'jpeg', 'png', 'jpg'
      * @return bool Success
      */
-    public static function extension($check, array $extensions = ['gif', 'jpeg', 'png', 'jpg']): bool
+    public static function extension(UploadedFileInterface|array|string $check, array $extensions = ['gif', 'jpeg', 'png', 'jpg']): bool
```

Comment:
Should functions like extension accept mixed also?

Reply:
Validation function take user input so it could be of any type.

Reply:
This is true. However, I've had to adjust only a few tests due to restricting types. I wonder if we're properly testing "supported" input in validation outside of a few specific exceptions.

Reply:
For example, this code certainly expects the type to be at least string and does not early return otherwise.

Reply:
I think so. Validation could be passed anything as it gets end user data.

Reply:
Updated extension() to mixed and  few others. Added a note in the docblock about supported values.